examples/us-dams/src/by-state.md

```diff
@@ -0,0 +1,339 @@
+    height: 50,
+    //marginTop: 40,
+    //marginBottom: 30,
+    color: { domain: conditions, range: conditionsColors, legend: true },
+    x: { label: "Number of dams" },
+    marks: [
+      Plot.barX(
+        conditionCounts,
+        Plot.stackX({
+          x: "count",
+          fill: "condition",
+          order: conditions,
+          tip: true,
+          rx: 2,
+          insetRight: 1
+        })
+      )
+    ],
+  });
+}
+```
+
+<!-- Create horizontal bar chart of dam types and ownership -->
+
+```js
+function purposeOwnership(width, height) {
+  return Plot.plot({
+    width,
+    marginTop: 0,
+    marginBottom: 40,
+    height: height - 55,
+    marginLeft: 240,
+    y: {label: null, label: "Primary purpose"},
+    x: {grid: true, label:"Number of dams"},
+    color: {legend: true, scheme: "Set2", domain: ["Private", "Public Utility", "Local Government", "State", "Federal"], label: "Ownership"},
+    marks:
+    [
+      Plot.barX(damsSelectedState, Plot.groupY({x: "count"}, {y: "primaryPurpose", rx: 2, insetRight: 1, sort: {y: "x", reverse: true, limit: 10}, fill: "ownerType", order: ["Private", "Public Utility", "Local Government", "State", "Federal"], tip: true}))
+    ]
+});
+}
+```
+
+<!-- Creates bubble chart of dam risk (hazard and condition) -->
+
+```js
+function conditionHeatmap(width, height) {
+  return Plot.plot({
+  width,
+  height,
+  marginRight: 10,
+  marginLeft: 100,
+  r: {range: [4, 20]},
+  y: {domain: ["Undetermined", "Low", "Significant", "High"], label: "Hazard potential", grid: true},
+  x: {domain: conditions, label: "Condition", grid: true, reverse: true},
+  color: {domain: conditions, range: conditionsColors, label: "Condition"},
+    marks: [
+    Plot.dot(damsSelectedState, Plot.group({r: "count"},
+      {y: "hazardPotential",
+      x: "conditionAssessment",
+      tip: true,
+      fill: "conditionAssessment"
+    }))
+  ]
+  });
+}
+```
+
+<!-- Create searchable table -->
+
+```js
+const damSearch = Inputs.search(damsSelectedState);
+
+const damSearchValue = Generators.input(damSearch);
+```
+
+<!-- Create interactived map with deck.gl -->
+
+```js
+import deck from "npm:deck.gl";
+
+const {DeckGL, AmbientLight, GeoJsonLayer, TextLayer, HexagonLayer, LightingEffect, PointLight, ScatterplotLayer} = deck;
+```
+
+```js
+// deck.gl setup
+const colorRange = [
+  [148,152,160], // not available
+  [66,105,208], // satisfactory
+  [151,187,245], // fair
+  [239, 213, 24], // unsatisfactory
+  [255, 114, 92] // poor
+];
+
+const colorLegend = Plot.plot({
+  margin: 0,
+  marginTop: 30,
+  marginRight: 20,
+  width: width / 4,
+  height: 50,
+  style: "color: 'currentColor';",
+  x: {padding: 0, axis: null},
+  marks: [
+    Plot.cellX(colorRange, {fill: ([r, g, b]) => `rgb(${r},${g},${b})`, inset: 0.5}),
+    Plot.text(["Fewer dams"], {frameAnchor: "top-left", dy: -12}),
+    Plot.text(["More dams"], {frameAnchor: "top-right", dy: -12})
+  ]
+});
+```
+
+```js
+const deckInstance = new DeckGL({
```

Comment:
for some reason this logs to the console:

`deck: Multiple new layers with same id scatter-plot`